django_cloud_deploy/cli/prompt.py

```diff
@@ -32,260 +33,430 @@
+class Command(enum.Enum):
+    NEW = 1
+    UPDATE = 2
 
-    @classmethod
-    @abc.abstractmethod
-    def prompt(cls,
-               console: io.IO,
-               step_prompt: str,
-               arguments: Dict[str, Any],
-               credentials: Optional[credentials.Credentials] = None) -> Any:
-        """Prompt the user to enter some information.
 
-        Args:
-            console: Object to use for user I/O.
-            step_prompt: A prefix showing the current step number e.g. "[1/3]".
-            arguments: The arguments that have already been collected from the
-                user e.g. {"project_id", "project-123"}
-            credentials: The OAuth2 Credentials object to use for api calls
-                during prompt.
+def _ask_prompt(question: str,
+                console: io.IO,
+                validate: Optional[Callable[[str], None]] = None,
+                default: Optional[str] = None) -> str:
+    """Used to ask for a single string value.
 
-        Returns:
-            The value entered by the user.
-        """
+    Args:
+        question: Question shown to the user on the console.
+        console: Object to use for user I/O.
+        validate: Function used to check if value provided is valid. It should
+            raise a ValueError if the the value fails to validate.
+        default: Default value if user provides no value. (Presses enter)
 
-    @staticmethod
-    def validate(s: str, credentials: Optional[credentials.Credentials] = None):
-        """Validates that a string is valid for this prompt type.
+    Returns:
+        The value entered by the user.
+    """
+    validate = validate or (lambda x: None)
+    while True:
+        answer = console.ask(question)
+        if default and not answer:
+            answer = default
+        try:
+            validate(answer)
+            break
+        except ValueError as e:
+            console.error(e)
+
+    return answer
+
+
+def _multiple_choice_prompt(question: str,
+                            options: List[str],
+                            console: io.IO,
+                            default: Optional[int] = None) -> Optional[int]:
+    """Used to prompt user to choose from a list of values.
+
+    Args:
+        question: Question shown to the user on the console. Should have
+            a {} to insert a list of enumerated options.
+        options: Possible values user should choose from.
+        console: Object to use for user I/O.
+        default: Default value if user provides no value. (Presses enter)
+
+    Typical usage:
+        # User can press enter if user doesn't want anything.
+        choice = _multiple_choice_prompt('Choose an option:\n{}\n',
+                                         ['Chicken', 'Salad', 'Burger'],
+                                         console,
+                                         default=None)
+
+    Returns:
+        The choice made by the user. Will be an index in the options list unless
```

Comment:
If there is no default set then shouldn't we force the user to choose (instead of returning `None`)? That's the way other prompts work.